### PR TITLE
Remove glog 

### PR DESCRIPTION
**Pull request by adrianchiris**
Remove the use of glog package.
use controller-runtime logging & zap instead.

- [x] Remove glog from cmds
- [x] Remove  glog from tests
- [x] Remove glog from pkgs (multiple packages, see below)
   - [x] webhook
  - [x] daemon
  - [x] host
  - [x] leaderelection
  - [x] plugins
  - [x] service
  - [x] systemd
  - [x] utils
- [x] remove glog flags from bindata/manifests 

roughly 460 substitutions required.

### CODE REVIEW

**Comment by adrianchiris** [comment]
@SchSeba @zeeke PTAL 

I have tested the change locally, PR rebased against master.

**Comment by adrianchiris** [comment]
Sample log from sriov config daemon:

```
2023-10-30T12:54:10.942768056Z  INFO    sriov-network-config-daemon     cobra/command.go:916    starting node writer
2023-10-30T12:54:10.962508147Z  INFO    sriov-network-config-daemon     cobra/command.go:916    Running on      {"platform": "Baremetal"}
2023-10-30T12:54:10.985544359Z  INFO    sriov-network-config-daemon/start.go:199        RunOnce()
2023-10-30T12:54:10.985637169Z  INFO    sriov-network-config-daemon/start.go:199        RunOnce(): first poll for nic status
2023-10-30T12:54:11.138692558Z  INFO    sriovnetwork    utils/utils.go:145      IsSupportedModel(): found unsupported model     {"vendorId:": "8086", "deviceId:": "1521"}
2023-10-30T12:54:11.138817271Z  INFO    daemon/writer.go:133    DiscoverSriovDevices(): unsupported device      {"device": "0000:01:00.0 -> driver: 'igb' class: 'Network controller' vendor: 'Intel Corporation' product: 'I350 Gigabit Network Connection'"}
2023-10-30T12:54:11.138928432Z  INFO    sriovnetwork    utils/utils.go:145      IsSupportedModel(): found unsupported model     {"vendorId:": "8086", "deviceId:": "1521"}
2023-10-30T12:54:11.138948937Z  INFO    daemon/writer.go:133    DiscoverSriovDevices(): unsupported device      {"device": "0000:01:00.1 -> driver: 'igb' class: 'Network controller' vendor: 'Intel Corporation' product: 'I350 Gigabit Network Connection'"}
2023-10-30T12:54:11.161606628Z  ERROR   utils/utils.go:163      getNetDevLinkSpeed(): fail to read Link Speed file      {"path": "/sys/class/net/enp129s0np0/speed", "error": "read /sys/class/net/enp129s0np0/speed: invalid argument"}
2023-10-30T12:54:11.168304929Z  INFO    daemon/writer.go:155    setNodeStateStatus(): status    {"sync-status": "", "last-sync-error": ""}
2023-10-30T12:54:11.190098177Z  INFO    daemon/writer.go:91     writeCheckpointFile(): try to decode the checkpoint file
2023-10-30T12:54:11.190385942Z  INFO    sriov-network-config-daemon     cobra/command.go:916    Starting SriovNetworkConfigDaemon
2023-10-30T12:54:11.190521775Z  INFO    sriov-network-config-daemon/start.go:220        Run()   {"node": "cloud-dev-44"}
2023-10-30T12:54:11.190566418Z  INFO    sriov-network-config-daemon/start.go:220        Run(): start daemon.
2023-10-30T12:54:11.190546985Z  INFO    runtime/asm_amd64.s:1598        Run(): start writer
2023-10-30T12:54:11.220722316Z  INFO    host/host.go:160        IsRHELSystem(): checking for RHEL machine
2023-10-30T12:54:11.220819078Z  INFO    host/host.go:171        IsUbuntuSystem(): checking for Ubuntu machine
2023-10-30T12:54:11.224171987Z  INFO    host/host.go:181        GetOSPrettyName(): getting os name from os-release file
2023-10-30T12:54:11.229748973Z  ERROR   daemon/daemon.go:217    tryEnableRdma(): failed to check OS name        {"error": "failed to find pretty operating system name"}
2023-10-30T12:54:11.229797893Z  INFO    host/host.go:127        LoadKernelModule(): try to load kernel module   {"name": "tun", "args": []}
2023-10-30T12:54:11.229823325Z  INFO    host/host.go:85 IsKernelModuleLoaded(): check if kernel module is loaded        {"name": "tun"}
2023-10-30T12:54:11.248953099Z  INFO    host/host.go:133        LoadKernelModule(): try to load kernel module   {"name": "vhost_net", "args": []}
2023-10-30T12:54:11.248996837Z  INFO    host/host.go:85 IsKernelModuleLoaded(): check if kernel module is loaded        {"name": "vhost_net"}
2023-10-30T12:54:11.297030525Z  INFO    daemon/daemon.go:429    Set log verbose level   {"new-level": -2, "current-level": 0}
2023-10-30T12:54:11.303536825Z  LEVEL(-2)       daemon/daemon.go:384    nodeUpdateHandler(): node is now drainable
2023-10-30T12:54:16.395449829Z  INFO    sriov-network-config-daemon/start.go:220        Starting workers
2023-10-30T12:54:16.395539681Z  INFO    sriov-network-config-daemon/start.go:220        Started workers
2023-10-30T12:54:16.395595008Z  LEVEL(-2)       daemon/daemon.go:319    processNextWorkItem     {"worker-queue-size": 1}
2023-10-30T12:54:16.395714341Z  LEVEL(-2)       daemon/daemon.go:319    get item from queue     {"item": 6}
2023-10-30T12:54:16.402562076Z  INFO    daemon/daemon.go:358    nodeStateSyncHandler(): new generation  {"generation": 6}
2023-10-30T12:54:16.402647605Z  INFO    runtime/asm_amd64.s:1598        Run(): refresh trigger
2023-10-30T12:54:16.402671302Z  LEVEL(-2)       daemon/writer.go:107    pollNicStatus()
2023-10-30T12:54:16.402726435Z  LEVEL(-2)       daemon/writer.go:133    DiscoverSriovDevices
2023-10-30T12:54:16.578223444Z  INFO    sriovnetwork    utils/utils.go:145      IsSupportedModel(): found unsupported model     {"vendorId:": "8086", "deviceId:": "1521"}
2023-10-30T12:54:16.57838351Z   INFO    daemon/writer.go:133    DiscoverSriovDevices(): unsupported device      {"device": "0000:01:00.0 -> driver: 'igb' class: 'Network controller' vendor: 'Intel Corporation' product: 'I350 Gigabit Network Connection'"}
2023-10-30T12:54:16.578532833Z  INFO    sriovnetwork    utils/utils.go:145      IsSupportedModel(): found unsupported model     {"vendorId:": "8086", "deviceId:": "1521"}
2023-10-30T12:54:16.578565673Z  INFO    daemon/writer.go:133    DiscoverSriovDevices(): unsupported device      {"device": "0000:01:00.1 -> driver: 'igb' class: 'Network controller' vendor: 'Intel Corporation' product: 'I350 Gigabit Network Connection'"}
2023-10-30T12:54:16.578702022Z  LEVEL(-2)       utils/utils.go:157      getNetdevMTU(): get MTU {"device": "0000:03:00.0"}
2023-10-30T12:54:16.57915546Z   LEVEL(-2)       utils/utils.go:162      getNetDevMac(): get Mac {"device": "enp3s0f0np0"}
2023-10-30T12:54:16.579250137Z  LEVEL(-2)       utils/utils.go:163      getNetDevLinkSpeed(): get LinkSpeed     {"device": "enp3s0f0np0"}
2023-10-30T12:54:16.579972537Z  LEVEL(-2)       utils/utils.go:165      getLinkType()   {"device": "0000:03:00.0"}
2023-10-30T12:54:16.583858403Z  LEVEL(-2)       utils/utils.go:179      GetNicSriovMode()       {"device": "0000:03:00.0"}
2023-10-30T12:54:16.585500736Z  LEVEL(-2)       utils/utils.go:712      getNetdevMTU(): get MTU {"device": "0000:03:00.2"}
...skipping...
2023-10-30T13:03:46.331949017Z  INFO    daemon/daemon.go:358    nodeStateSyncHandler(): new generation  {"generation": 6}
2023-10-30T13:03:46.332037677Z  INFO    daemon/daemon.go:358    nodeStateSyncHandler(): Interface not changed
2023-10-30T13:03:46.332061748Z  INFO    daemon/daemon.go:374    Successfully synced
2023-10-30T13:03:53.920737008Z  INFO    sriovnetwork    utils/utils.go:145      IsSupportedModel(): found unsupported model     {"vendorId:": "8086", "deviceId:": "1521"}
2023-10-30T13:03:53.920888959Z  INFO    daemon/writer.go:133    DiscoverSriovDevices(): unsupported device      {"device": "0000:01:00.0 -> driver: 'igb' class: 'Network controller' vendor: 'Intel Corporation' product: 'I350 Gigabit Network Connection'"}
2023-10-30T13:03:53.921027012Z  INFO    sriovnetwork    utils/utils.go:145      IsSupportedModel(): found unsupported model     {"vendorId:": "8086", "deviceId:": "1521"}
2023-10-30T13:03:53.921060772Z  INFO    daemon/writer.go:133    DiscoverSriovDevices(): unsupported device      {"device": "0000:01:00.1 -> driver: 'igb' class: 'Network controller' vendor: 'Intel Corporation' product: 'I350 Gigabit Network Connection'"}
2023-10-30T13:03:53.950043033Z  ERROR   utils/utils.go:163      getNetDevLinkSpeed(): fail to read Link Speed file      {"path": "/sys/class/net/enp129s0np0/speed", "error": "read /sys/class/net/enp129s0np0/speed: invalid argument"}
2023-10-30T13:03:53.957377258Z  INFO    daemon/writer.go:155    setNodeStateStatus(): status    {"sync-status": "Succeeded", "last-sync-error": ""}
2023-10-30T13:04:01.331030241Z  INFO    daemon/daemon.go:358    nodeStateSyncHandler(): new generation  {"generation": 6}
2023-10-30T13:04:01.331133148Z  INFO    daemon/daemon.go:358    nodeStateSyncHandler(): Interface not changed
2023-10-30T13:04:01.331158258Z  INFO    daemon/daemon.go:374    Successfully synced
```

**Review by SchSeba — APPROVED** [review]
looks great thanks for the effort!

/lgtm

**Comment by bn222** [comment]
LGTM

Question, why is the old log package still an indirect dependency?

**Comment by adrianchiris** [comment]
> Question, why is the old log package still an indirect dependency?

maybe because of dependency on sriov device plugin ?

File: pkg/daemon/daemon.go [hunk 1]
```diff
@@ -424,15 +426,22 @@ func (dn *Daemon) operatorConfigAddHandler(obj interface{}) {
...
 	if dn.disableDrain != newDisableDrain {
 		dn.disableDrain = newDisableDrain
-		glog.Infof("Set Disable Drain to: %t", dn.disableDrain)
+		log.Log.Info("Set Disable Drain", "value", dn.disableDrain)
+	}
+}
+
+// handleLogLevelChange handles log level change
+func (dn *Daemon) handleLogLevelChange(logLevel int) {
+	newLevel := int8(logLevel * -1)
```
Comment (mlguerrero12) [inline pkg/daemon/daemon.go:440]: perhaps you can use log.Log.V(logLevel).Enabled() to verify whether the new level is the same or not. You won't need to have a reference to the zap package.

**Comment by mlguerrero12** [comment]
Too late for comments. Already merged.

**Pull request merged**